Python/Product/TestAdapter/visualstudio_py_testlauncher.py

```diff
@@ -17,23 +17,174 @@
+                test = test.id()
+            )
+
+    def addError(self, test, err):
+        super(VsTestResult, self).addError(test, err)
+        self.sendResult(test, 'failed', err)
+
+    def addFailure(self, test, err):
+        super(VsTestResult, self).addFailure(test, err)
+        self.sendResult(test, 'failed', err)
+
+    def addSuccess(self, test):
+        super(VsTestResult, self).addSuccess(test)
+        self.sendResult(test, 'passed')
+
+    def addSkip(self, test, reason):
+        super(VsTestResult, self).addSkip(test, reason)
+        self.sendResult(test, 'skipped')
+
+    def addExpectedFailure(self, test, err):
+        super(VsTestResult, self).addExpectedFailure(test, err)
+        self.sendResult(test, 'failed', err)
+
+    def addUnexpectedSuccess(self, test):
+        super(VsTestResult, self).addUnexpectedSuccess(test)
+        self.sendResult(test, 'passed')
+
+    def sendResult(self, test, outcome, trace = None):
+        if _channel is not None:
+            tb = None
+            message = None
+            if trace is not None:
+                traceback.print_exc()
+                formatted = traceback.format_exception(*trace)
+                # Remove the 'Traceback (most recent call last)'
+                formatted = formatted[1:]
+                tb = ''.join(formatted)
+                message = str(trace[1])
+            _channel.send_event(
+                name='result', 
+                outcome=outcome,
+                traceback = tb,
+                message = message,
+                test = test.id()
+            )
+
 def main():
     import os
     import sys
     import unittest
     from optparse import OptionParser
-    
+    global _channel
+
     parser = OptionParser(prog = 'visualstudio_py_testlauncher', usage = 'Usage: %prog [<option>] <test names>... ')
     parser.add_option('-s', '--secret', metavar='<secret>', help='restrict server to only allow clients that specify <secret> when connecting')
     parser.add_option('-p', '--port', type='int', metavar='<port>', help='listen for debugger connections on <port>')
     parser.add_option('-x', '--mixed-mode', action='store_true', help='wait for mixed-mode debugger to attach')
     parser.add_option('-t', '--test', type='str', dest='tests', action='append', help='specifies a test to run')
-    parser.add_option('-m', '--module', type='str', help='name of the module to import the tests from')
-    
+    parser.add_option('-c', '--coverage', type='str', help='enable code coverage and specify filename')
+    parser.add_option('-r', '--result-port', type='int', help='connect to port on localhost and send test results')
     (opts, _) = parser.parse_args()
     
     sys.path[0] = os.getcwd()
     
+    if opts.result_port:
+        _channel = _IpcChannel(socket.create_connection(('127.0.0.1', opts.result_port)))
+        sys.stdout = _TestOutput(sys.stdout, is_stdout = True)
+        sys.stderr = _TestOutput(sys.stderr, is_stdout = False)
```

Comment:
Might be better to pass in a `name` argument anyway. Certainly a nicer looking API